### PR TITLE
Add missing Rotation functions

### PR DESCRIPTION
**Pull request by chrisflesher**
Added some missing functions so the API matches scipy now:

```
align_vectors
create_group
random
reduce
```

Also fixed a bug in from_euler when only one axis is specified.

This PR is the continuation PR #17192 but based on the latest `main` branch.

### CODE REVIEW

**Comment by chrisflesher** [comment]
Hi @jonesk7734 could you trigger CI for this please? Hopefully it is ready now...

**Comment by chrisflesher** [comment]
Hi @jakevdp could you trigger CI for this please? Hopefully it is ready now...

**Comment by chrisflesher** [comment]
I wasn't able to get this PR accepted so have been making updates in a separate repo:

https://github.com/chrisflesher/jax-scipy-spatial

I've written it to be easy to merge back into the JAX repo if there's any desire to do so.

**Comment by jakevdp** [comment]
Hi - sorry we lost track of this. The rotation functions are now considered [out-of-scope](https://jax.readthedocs.io/en/latest/jep/18137-numpy-scipy-scope.html#scipy-spatial) for `jax.scipy`, so we likely will not be adding additional functionality to them.

A separate package looks useful, but a better approach may be to support jax rotation operations via scipy's array API support (see https://github.com/scipy/scipy/issues/20544). JAX is close to having full support for the Array API (see #21088), and once `scipy.spatial` adds array API support those functions will be able to be used directly on JAX arrays, and lower to JAX operations. At that point, we will likely deprecate and remove `jax.scipy.spatial`.

**Comment by chrisflesher** [comment]
Ah, sounds great! Thanks for letting me know about the Array API changes, I was planning to manually convert more scipy libraries but think this will free up some time to work on something else.

**Comment by mdhaber** [comment]
I think the best existing issue for adding array API support to `scipy.spatial` is scipy/scipy#18867. It might be worth creating a more specific issue for `spatial` similar to scipy/scipy#20544 (which is specific to `stats`).
